src/adaptors/in-memory-articles.ts

```diff
@@ -0,0 +1,33 @@
+import { JsonLdObj, Iri } from 'jsonld/jsonld-spec';
+import { Articles } from './articles';
+
+export default class InMemoryArticles implements Articles {
+    private nodes: { [key: string]: JsonLdObj } = {};
+
+    * [Symbol.iterator](): Iterator<JsonLdObj> {
```

Comment:
That C feeling (I know, [not a pointer](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/yield*))